iml-agent/src/action_plugins/stratagem/action_filesync.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+extern crate futures;
+extern crate tokio;
+
+use crate::agent_error::{ImlAgentError, RequiredError};
+use crate::lustre::search_rootpath;
+use futures::{
+    future::{self, TryFutureExt},
+    stream::{self, StreamExt, TryStreamExt},
+};
+use iml_wire_types::{FidError, FidItem};
+use liblustreapi::LlapiFid;
+use std::{collections::HashMap, path::PathBuf};
+use tokio::fs;
+use tokio::process::Command;
+
+struct Work {
+    src_file: String,
+    dest_file: String,
+}
+
+async fn actually_rsync(work: &Work) -> Result<(), std::io::Error> {
+    tracing::error!("running {}", work.src_file);
+    let output = Command::new("rsync")
+        .arg("-a")
+        .arg("-t")
+        .arg("-X")
+        .arg(&work.src_file)
+        .arg(&work.dest_file)
+        .output();
+
+    let output = output.await?;
+
+    tracing::debug!("done {} {}", work.src_file, output.status);
+
+    Ok(())
+}
+
+async fn do_rsync(work_list: &Vec<Work>) -> Result<Vec<FidError>, ImlAgentError> {
```

Comment:
This fn is being treated as infallible from the outside (all errors are being printed instead of bubbled). Because of that we can eliminate the return of a `Result` and just do this:

```suggestion
async fn do_rsync(work_list: &[Work]) {
    stream::iter(work_list)
        .for_each_concurrent(100, |fids| async move {
            match actually_rsync(fids).await {
                Ok(_) => tracing::debug!("All done"),
                Err(e) => tracing::warn!("Error: {}", e),
            };
        })
        .await;
}
```